### PR TITLE
Update openshift to 4.4. and k8s to 1.17

### PR DESCRIPTION
**Pull request by kadel**
**What type of PR is this?**
/kind dependency-change

**What does this PR do / why we need it**:
- updates OpenShift libraries to 4.4 and k8s libs to 4.4
- This is required to be able to add additional functionality that is not available in the previous versions
- replaced glog with klog everywhere (https://github.com/kubernetes/klog) glog is not actively maintained anymore

**Which issue(s) this PR fixes**:
Fixes https://github.com/openshift/odo/issues/3077

**How to test changes / Special notes**
just build and run odo, it should work as before

### CODE REVIEW

File: cmd/odo/odo.go [hunk 1]
```diff
@@ -77,7 +79,7 @@ func main() {
 		case message := <-updateInfo:
 			log.Italic(message)
 		default:
-			glog.V(4).Info("Could not get the latest release information in time. Never mind, exiting gracefully :)")
+			klog.V(4).Info("Could not get the latest release information in time. Never mind, exiting gracefully :)")
```
Comment (mik-dass) [inline cmd/odo/odo.go:82]: why are we moving to klog instead of glog?

Comment (kadel) [reply cmd/odo/odo.go:82]: klog is kubernetes fork of glog. Glog is not maintained anymore.

https://github.com/kubernetes/klog

**Comment by mik-dass** [comment]
https://travis-ci.com/github/openshift/odo/jobs/328172115#L723

It seems to fail because we are not providing the two required flags

**Comment by mik-dass** [comment]
@kadel I can't even load the 30k+ file changes on the browser. Retried a couple of times :D

**Comment by kadel** [comment]
> https://travis-ci.com/github/openshift/odo/jobs/328172115#L723
> 
> It seems to fail because we are not providing the two required flags

The new version of Cobra changed the capitalization of the first letter in the error message. This is why the test was failing

File: cmd/odo/odo.go (hunk 1 above)
Comment (mik-dass) [reply cmd/odo/odo.go:82]: :+1:

**Comment by mik-dass** [comment]
> The new version of Cobra changed the capitalization of the first letter in the error message. This is why the test was failing

OK :+1: I guess I missed it.

**Pull request merged**